src/neo4j_genai/retrievers/external/weaviate/weaviate.py

```diff
@@ -0,0 +1,138 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import Optional
+from neo4j_genai.retrievers.base import ExternalRetriever
+from neo4j_genai.embedder import Embedder
+from neo4j_genai.types import SearchType
+from neo4j_genai.retrievers.utils import validate_search_query_input
+import weaviate.classes as wvc
+import neo4j
+import logging
+from neo4j_genai.neo4j_queries import get_search_query
+
+logger = logging.getLogger(__name__)
+
+
+class WeaviateNeo4jRetriever(ExternalRetriever):
+    def __init__(
+        self,
+        driver: neo4j.Driver,
+        client: any,
+        collection: str,
+        id_property_external: str,
+        id_property_neo4j: str = None,
+        embedder: Optional[Embedder] = None,
+        return_properties: Optional[list[str]] = None,
+        retrieval_query: Optional[str] = None,
+    ):
+        self.driver = driver
+        self.client = client
+        self.search_collection = client.collections.get(collection)
+        self._id_property_external = id_property_external
+        self._id_property_neo4j = id_property_neo4j
+        self.embedder = embedder
+        self.return_properties = return_properties
+        self.retrieval_query = retrieval_query
+
+    @property
+    def id_property_external(self):
+        return self._id_property_external
+
+    @property
+    def id_property_neo4j(self):
+        return self._id_property_neo4j
+
+    def search(
```

Comment:
Indeed, if we add filters here, which makes sense, I think it can be weaviate format, not sure it makes sense to support converting from "our" format to another one here, since we'd have to deal with updates and so on.
The only advantage (?) of our format, is when we need to translate it to Cypher, which is not the case with external providers.

Reply:
Yeah, I agree with this